packages/cli/src/ActiveExecutions.ts

```diff
@@ -61,30 +58,26 @@ export class ActiveExecutions {
 				fullExecutionData.workflowId = workflowId;
 			}
 
-			const execution = ResponseHelper.flattenExecutionData(fullExecutionData);
-
-			const executionResult = await Db.collections.Execution.save(execution as IExecutionFlattedDb);
+			const executionResult = await Container.get(ExecutionRepository).createNewExecution(
```

Comment:
instead of get()ing the service twice, could we not inject it into the ActiveExecutions through its constructor (since both are going to be only instantiated once if i am not mistaken?)

Reply:
I've tried that but for some reason the Container was unable to actually inject the repository. Maybe this is due to some race conditions while initialising of that the DB class gets loaded after this one, in any case, I talked with Adi and it would be a bit more complicated, so I've left it out for now.

Reply:
fair enough. it's interesting though since as i understand it, container.get() should instantiate the service if it does not exist yet... hm.

Reply:
Yup that's weird, but the error I was getting is that the service was not found. Oddly, using `Container.get()` it works because at some point during init, the Db class is initialised and the services "register" themselves, so then the `get` call works.

Reply:
stupid question: when you were instantiating ActiveExecutions in the version where ExecutionRepository was in the constructor, you were passing it an instance of ExecutionRepository created by Container.get(), yes?